sitehost/dns/resource.go

```diff
@@ -0,0 +1,296 @@
+	resp, err := client.GetRecord(ctx, dns.RecordRequest{
+		ID:         d.Id(),
+		DomainName: domain,
+	})
+	if err != nil {
+		return diag.Errorf("Error retrieving DNS zone: %s", err)
+	}
+
+	if err := setRecordAttributes(d, resp); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func deleteRecordResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf, ok := meta.(*helper.CombinedConfig)
+	if !ok {
+		return diag.Errorf("failed to convert meta object")
+	}
+
+	client := dns.New(conf.Client)
+	resp, err := client.DeleteRecord(ctx, dns.DeleteRecordRequest{
+		Domain:   fmt.Sprintf("%v", d.Get("domain")),
+		RecordID: d.Id(),
+	})
+	if err != nil {
+		return diag.Errorf("Error deleting DNS record: %s", err)
+	}
+
+	if !resp.Status {
+		return diag.Errorf("Error deleting DNS record: %s", resp.Msg)
+	}
+
+	return nil
+}
+
+func updateRecordResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf, ok := meta.(*helper.CombinedConfig)
+	if !ok {
+		return diag.Errorf("failed to convert meta object")
+	}
+
+	client := dns.New(conf.Client)
+	resp, err := client.UpdateRecord(
+		ctx,
+		dns.UpdateRecordRequest{
+			Domain:   fmt.Sprintf("%v", d.Get("domain")),
+			RecordID: d.Id(),
+			Type:     fmt.Sprintf("%v", d.Get("type")),
+			Name:     fmt.Sprintf("%v", d.Get("name")),
+			Content:  fmt.Sprintf("%v", d.Get("record")),
+			Priority: fmt.Sprintf("%v", d.Get("priority")),
+		},
+	)
+	if err != nil {
+		return diag.Errorf("Error updating DNS record: %s", err)
+	}
+
+	if !resp.Status {
+		return diag.Errorf("Error updating DNS record: %s", resp.Msg)
+	}
+
+	record, err := client.GetRecord(ctx, dns.RecordRequest{
+		ID:         d.Id(),
+		DomainName: fmt.Sprintf("%v", d.Get("domain")),
+	})
+	if err != nil {
+		return diag.Errorf("Error creating DNS record: %s", err)
+	}
+
+	if err := setRecordAttributes(d, record); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func importRecordResource(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	if strings.Contains(d.Id(), ",") {
+		s := strings.Split(d.Id(), ",")
+
+		d.SetId(s[1])
+		if err := d.Set("domain", s[0]); err != nil {
+			return nil, err
+		}
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func setRecordAttributes(d *schema.ResourceData, record models.DNSRecord) error {
+	d.SetId(record.ID)
+
+	if err := d.Set("domain", record.Domain); err != nil {
+		return err
+	}
+
+	if err := d.Set("name", record.Name); err != nil {
+		return err
+	}
+
+	if err := d.Set("type", record.Type); err != nil {
+		return err
+	}
+
+	priority, err := strconv.Atoi(record.Priority)
+	if err != nil {
+		priority = 0
+	}
+
+	if err := d.Set("priority", priority); err != nil {
+		return err
+	}
+
+	if err := d.Set("record", record.Content); err != nil {
+		return err
+	}
+
+	if err := d.Set("change_date", record.ChangeDate); err != nil {
```

Comment:
You can just do `return d.Set("change_date", record.ChangeDate)` and remove everything after because either way you return error or nil.